src/main/java/vct/col/newrewrite/SpecifyImplicitLabels.scala

```diff
@@ -0,0 +1,41 @@
+package vct.col.newrewrite
+
+import hre.util.ScopedStack
+import vct.col.ast.RewriteHelpers._
+import vct.col.ast._
+
+case class ImplicitLabelOrigin(inner: Origin) extends Origin {
+  override def preferredName: String = "implicitLabel"
+  override def messageInContext(message: String): String = inner.messageInContext(message)
+}
+
+case class SpecifyImplicitLabels() extends Rewriter {
+  val labelStack = new ScopedStack[LabelDecl]()
+
+  def isBreakable(s: Statement) = s match {
+    case _: Loop => true
+    case _: Switch => true
+    case _ => false
+  }
+
+  override def dispatch(stat: Statement): Statement = stat match {
+    case block@Block(Seq(oldLabel@Label(_), s: Statement)) if isBreakable(s) =>
+      val newLabel = oldLabel.rewrite()
+      val newS = labelStack.having(newLabel.decl) {
+        rewriteDefault(s)
```

Comment:
dispatch

Reply:
In this case I want the case on line 28 not to trigger if the case on line 22 is triggered, so here I think it _should_ be `rewriteDefault`...?